### PR TITLE
Start design vignette

### PR DESCRIPTION
**Pull request by Bisaloo**
- [x] I have read the CONTRIBUTING guidelines 
- [x] A new item has been added to `NEWS.md`

### CODE REVIEW

File: vignettes/design-principles.Rmd [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+## Scope
+
+linelist provides a lightweight layer to add tags to data.frame columns. This allows:
+
+- column identification without renaming
+- extra feature for the tagged columns, such as the ability to warn when a tagged column is dropped, or when its data type is incompatible with the expected one.
+
+## Input/Output/Interoperability
+
+Because of its scope, linelist is intended to provide maximum compatibility with data.frames, or packages defining subclasses of data.frames.
```
Comment (joshwlambert) [inline vignettes/design-principles.Rmd:30]: I assume the position of {linelist} is that any difference of behaviour between `<data.frame>` subclasses should be understood by the user and is not the concern of {linelist} which is just wrapping? E.g. https://tibble.tidyverse.org/articles/wip/subset.html.

File: vignettes/design-principles.Rmd [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+We prefer not adding a new feature, rather than having this feature alter the usual behaviour of a data.frame.
+
+`make_linelist()` is the main user-facing function of this package. It takes a data.frame/tibble/X as input as returns an output of class `c("linelist", "data.frame")`/`c("linelist", "tibble")`/`c("linelist", "X")`.
```
Comment (joshwlambert) [inline vignettes/design-principles.Rmd:34]: Monospace "data.frame/tibble/X"?

Comment (joshwlambert) [reply vignettes/design-principles.Rmd:34]: Worth adding here that `<linelist>` is incompatible with `<data.table>`?

File: vignettes/design-principles.Rmd [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+## Design decisions
+
+- Wherever possible, linelist should not provide its own method, but degrade gracefully and rely on the superclass method. This is to ensure that linelist is as compatible as possible with other packages and data types.
+
+## Dependencies
+
+Because of its strong interoperability with the tidyverse package, it is accepted for linelist to depend on low-level tidyverse or r-lib packages, such as rlang, vctrs or tidyselect.
```
Comment (joshwlambert) [inline vignettes/design-principles.Rmd:42]: "tidyverse packages" plural?

Comment (joshwlambert) [reply vignettes/design-principles.Rmd:42]: Might be nice to hyperlink each example tidyverse package to their pkgdown site, e.g. https://rlang.r-lib.org/index.html.

File: vignettes/design-principles.Rmd [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignette outlines the design decisions that have been taken during the development of the `{linest}` R package, and provides some of the reasoning, and possible pros and cons of each decision.
```
Comment (joshwlambert) [inline vignettes/design-principles.Rmd:17]: Remove {} from package name for consistency with rest of document? I think the backticks help distinguish it as a package name so could also use them for other "linelist" occurrences.

File: vignettes/design-principles.Rmd [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+This document is primarily intended to be read by those interested in understanding the code within the package and for potential package contributors.
+
+## Scope
+
+linelist provides a lightweight layer to add tags to data.frame columns. This allows:
```
Comment (joshwlambert) [inline vignettes/design-principles.Rmd:23]: Backticks for "data.frame"? Possibly also `<>` class notation?

**Review by joshwlambert — APPROVED** [review]
@Bisaloo thanks for tagging. It looks good to me. As with other design vignettes it can change organically with updates to the package.

File: vignettes/design-principles.Rmd (hunk 5 above)
Comment (Bisaloo) [reply vignettes/design-principles.Rmd]: Fixed in [`ee556b4` (#112)](https://github.com/epiverse-trace/linelist/pull/112/commits/ee556b4a74777236141baf3fbda35388a2f5a6d1).

File: vignettes/design-principles.Rmd (hunk 4 above)
Comment (Bisaloo) [reply vignettes/design-principles.Rmd]: Fixed in [`ee556b4` (#112)](https://github.com/epiverse-trace/linelist/pull/112/commits/ee556b4a74777236141baf3fbda35388a2f5a6d1).

File: vignettes/design-principles.Rmd (hunk 1 above)
Comment (Bisaloo) [reply vignettes/design-principles.Rmd:30]: Yes, it's now explicitly called out via [`76926a7` (#112)](https://github.com/epiverse-trace/linelist/pull/112/commits/76926a7e9ec471a423d4660ab1bd5d3e4ae3e24d).

File: vignettes/design-principles.Rmd (hunk 2 above)
Comment (Bisaloo) [reply vignettes/design-principles.Rmd]: Fixed by [`ee556b4` (#112)](https://github.com/epiverse-trace/linelist/pull/112/commits/ee556b4a74777236141baf3fbda35388a2f5a6d1) and [`385e18d` (#112)](https://github.com/epiverse-trace/linelist/pull/112/commits/385e18dcf2ce3bf536329cb24b85d468557783f7)

File: vignettes/design-principles.Rmd (hunk 3 above)
Comment (Bisaloo) [reply vignettes/design-principles.Rmd]: Fixed by [`ee556b4` (#112)](https://github.com/epiverse-trace/linelist/pull/112/commits/ee556b4a74777236141baf3fbda35388a2f5a6d1).

**Pull request merged**